src/Workspaces/CoreTest/Remote/ServiceDescriptorTests.cs

```diff
@@ -153,6 +154,19 @@ public void TypesUsedInRemoteApisMustBeMessagePackSerializable()
                     {
                         errors.Add($"{type} referenced by {declaringMember} is an internal enum and needs a custom formatter");
                     }
+                    else if (type.IsAbstract)
+                    {
+                        // custom abstract types must be explicitly listed in MessagePackFormatters.AbstractTypeFormatters
+                        if (!MessagePackFormatters.Formatters.Any(
+                            formatter => formatter.GetType() is { IsGenericType: true } and var formatterType &&
+                                         formatterType.GetGenericTypeDefinition() == typeof(ForceTypelessFormatter<>) &&
+                                         formatterType.GenericTypeArguments[0] == type))
+                        {
+                            errors.Add($"{type} referenced by {declaringMember} is abstract but ForceTypelessFormatter<{type}> is not listed in {nameof(MessagePackFormatters)}.{nameof(MessagePackFormatters.Formatters)}");
+                        }
```

Comment:
part of this PR?

Reply:
Yes, turns out serialization of abstract classes doesn't "just work".